docling_ibm_models/layoutmodel/layout_predictor.py

```diff
@@ -76,34 +72,50 @@ def __init__(
         self._threshold = 0.6  # Score threshold
         self._image_size = 640
         self._size = np.asarray([[self._image_size, self._image_size]], dtype=np.int64)
-        self._use_cpu_only = use_cpu_only or ("USE_CPU_ONLY" in os.environ)
+
+        # Set device based on env var or availability
+        device_name = os.environ.get("TORCH_DEVICE", "").lower()
+        if device_name in ["cuda", "mps", "cpu"]:
+            self.device = torch.device(device_name)
```

Comment:
what happens if this variable is set to a non-available device? e.g. cuda on macos, or mps on linux?